corehq/apps/reports/formdetails/case_metadata.py

```diff
@@ -194,41 +187,36 @@ def _add_advanced_form_contribution(self, form):
             if action.close_condition.is_active():
                 meta.add_closer(form.unique_id, action.close_condition)
 
-    def _add_save_to_case_questions(self, form, form_questions):
-        def _make_save_to_case_question(path):
-            # todo: this is a hack - just make an approximate save-to-case looking question
-            return FormQuestionResponse.wrap({
-                "label": path,
-                "tag": path,
-                "value": path,
-                "repeat": None,
-                "group": None,
-                "type": 'SaveToCase',
-                "relevant": None,
-                "required": None,
-                "comment": None,
-                "hashtagValue": path,
-            })
+    def _add_save_to_case_questions(self, form):
+        # The information for save to case questions comes directly from Vellum
+        for property_info in form.case_references_data.get_save_references():
+            if not property_info.case_type:
```

Comment:
This is a nice readability improvement (switching to a `continue` so indentation is reduced and adding an explanatory comment).